### DISCUSSION TITLE
Reflector supporting both audio and video

### DISCUSSION THREAD

**Discussion by MartinPulec** [CESNET/UltraGrid#388] [Ideas]
> Also wondering why it would need that run twice for both audio and video, wouldn't be easier to put that all into one?

_Originally posted by @TheSashmo in https://github.com/CESNET/UltraGrid/issues/379#issuecomment-2030648631_

**Comment by alatteri** [comment]
I agree here. Overall, having Video and Audio packetized together at all steps would make a few things simpler.

**Comment by MartinPulec** [comment]
It has some historic connotations, mainly that as the audio and video uses separate port, the reflector, originally entirely content agnostic (which is no longer true for the transcoding reflector) retransmits one port.

>  wouldn't be easier to put that all into one?

From the user perspective I believe that certainly. From the development it is more tricky, because if using the (video) transcoding reflector, there are now added much more features, like the subscription server mode etc. which I am not sure if we are ready to support now.

Anyways, would it help if we just create a wrapper script that would start the audio and video reflector simultaneously? My idea was to keep the options eg. for video recompression possible also for the wrapper, but passing them just to the video reflector, not the audio.

**Comment by alatteri** [comment]
I've already done that myself for our implementation.  I'd rather see your time go towards other development features, instead of something that i consider to be more of a local use case-specific implementation.

> Anyways, would it help if we just create a wrapper script that would start the audio and video reflector simultaneously? My idea was to keep the options eg. for video recompression possible also for the wrapper, but passing them just to the video reflector, not the audio.

**Comment by MartinPulec** [comment]
Well, I've done one in the end:

[hd-rum-av](https://github.com/CESNET/UltraGrid/blob/master/data/template/bin/hd-rum-av)

I believe it is a good idea - it simplifies things (you don't need to have a terminal multiplexor or multiple terminals); also it handles things like [videoconferencing mode](https://github.com/CESNET/UltraGrid/wiki/Videoconferencing-Mode) without the need to start both processes (which have entirely different command).
